File: java-example/src/main/java/no/javatec/pubsubemulator/spock/controller/DemoController.java
package no.javatec.pubsubemulator.spock.controller;

import io.micronaut.http.MediaType;
import io.micronaut.http.annotation.*;
import io.micronaut.scheduling.TaskExecutors;
import io.micronaut.scheduling.annotation.ExecuteOn;
import no.javatec.pubsubemulator.spock.dto.SampleInputMessage;
import no.javatec.pubsubemulator.spock.dto.SampleReturnMessage;
import no.javatec.pubsubemulator.spock.service.ControllerService;

@ExecuteOn(TaskExecutors.IO)
@Controller("/pubsubEmulatorDemo")
public class DemoController {

    private final ControllerService demoService;

    public DemoController(ControllerService demoService) {
        this.demoService = demoService;
    }

    @Produces(MediaType.TEXT_PLAIN)
    @Get
    public String index() {
        return "Example Response";
    }

    @Post
    public SampleReturnMessage postMethod(@Body SampleInputMessage inputMessage) {
        return demoService.processRequest(inputMessage);
    }
}

File: kotlin-example/src/main/kotlin/no/javatec/pubsubemulator/kotest/controller/DemoController.kt
package no.javatec.pubsubemulator.kotest.controller

import io.micronaut.http.MediaType
import io.micronaut.http.annotation.*
import io.micronaut.scheduling.TaskExecutors
import io.micronaut.scheduling.annotation.ExecuteOn
import no.javatec.pubsubemulator.kotest.dto.SampleInputMessage
import no.javatec.pubsubemulator.kotest.service.ControllerService

@ExecuteOn(TaskExecutors.IO)
@Controller("/pubsubEmulatorDemo")
class DemoController(private val demoService: ControllerService) {

    @Produces(MediaType.TEXT_PLAIN)
    @Get
    fun index(): String = "Example Response"

    @Post
    fun postMethod(@Body inputMessage: SampleInputMessage) = demoService.processRequest(inputMessage)
}